tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/RabbitMQ.cs

```diff
@@ -206,10 +221,47 @@ void handler(object ch, BasicDeliverEventArgs basicDeliverEventArgs)
 #else
                     receivedMessage = Encoding.UTF8.GetString(basicDeliverEventArgs.Body);
 #endif
+                    VerifyHeaders(basicDeliverEventArgs.BasicProperties.Headers);
 
                     manualResetEvent.Set();
                 }
             }
         }
+
+        private void VerifyHeaders(IDictionary<string, object> headers)
```

Comment:
I realize this code is pretty gross and I'd appreciate any ideas for how to make it cleaner.

Reply:
Looks like this is because the header strings get encoded as bytes, so you might be able to simplify a little by only checking that case and falling back to `.Equals()` otherwise.

Reply:
I feel like the code is complicated because it attempts to future-proof additions to the headers collection instead of just using known keys/values test cases. I'm not sure I would refactor it to not be future proof though. 👍

Reply:
I took @tehbio's advice and that simplified things a bit, take a look.